libpod/runtime_img.go

```diff
@@ -106,11 +106,17 @@ func (r *Runtime) RemoveImage(ctx context.Context, image *image.Image, force boo
 		}
 	}
 
-	if len(image.Names()) > 1 && !image.InputIsID() {
+	storageLayers, err := r.GetLayers()
+	if err != nil {
+		return "", errors.Wrap(err, "error getting layers from store")
+	}
+	hasChildren := r.IsParent(storageLayers, image.TopLayer())
+
+	if (len(image.Names()) > 1 && !image.InputIsID()) || hasChildren {
```

Comment:
So, keeping in mind that I'm fairly ignorant of both Go and this code base... I'm wondering if this is really the most appropriate logic for the desired effect? Instead of overloading this conditional and `MatchRepoTag`, might it make sense to add something like this just after the `hasChildren` definition on line 113 above?

    if image.InputIsId() && hasChildren {
        return "", "unable to remove image (cannot be forced) - has children"
    }

If my thinking is correct, that lets you leave `MatchRepoTag` untouched. I think you'd need to change the `> 1` to `>= 1` here, and possibly break out of the `if` (instead of `return "Untagged"`) if image.Names == 1, but even so it seems a little more natural to me. (But again: take this with huge grain of salt).